ckan/logic/action/update.py

```diff
@@ -777,10 +783,151 @@ def _group_or_org_update(context, data_dict, is_org=False):
 
     if not context.get('defer_commit'):
         model.repo.commit()
+        # (canada fork only): background search index rebuilding
+        #TODO: upstream contrib!!
+        if (plugins.toolkit.asbool(plugins.toolkit.config.get('ckan.search.reindex_after_group_update', False))
+            and (group.title != current_title or group.name != current_name)):
+            # only reindex if the title or name has changed
+            action = 'reindex_group_datasets'
+            if is_org:
+                action = 'reindex_organization_datasets'
+            reindex_context = {
+                'model': model,
+                'user': user,
+                'ignore_auth': True,
+                'session': session
+            }
+            _get_action(action)(reindex_context, {'id': group.id})
 
     return model_dictize.group_dictize(group, context)
 
 
+# (canada fork only): background search index rebuilding
+#TODO: upstream contrib!!
+def _group_or_org_reindex(context, data_dict, is_org=False):
+    model = context['model']
+    session = context['session']
+    id = _get_or_bust(data_dict, 'id')
+
+    group = model.Group.get(id)
+    context["group"] = group
+    if group is None:
+        raise NotFound('Group was not found.')
+
+    data_dict['type'] = group.type
+
+    if is_org:
+        _check_access('reindex_organization_datasets', context, data_dict)
+    else:
+        _check_access('reindex_group_datasets', context, data_dict)
+
+    results = session.query(model.Package.id)\
+        .filter(model.Package.owner_org == group.id)\
+        .filter(model.Package.state == 'active').all()
+
+    if not results:
+        return
+
+    task = {
+        'entity_id': group.id,
+        'entity_type': 'group',
+        'task_type': 'search_rebuild',
+        'last_updated': str(datetime.datetime.now(datetime.timezone.utc)),
+        'state': 'submitting',
+        'key': 'search_rebuild',
+        'value': '{}',
+        'error': '{}',
+    }
+    try:
+        existing_task = _get_action('task_status_show')(context, {'entity_id': group.id,
+                                                                  'task_type': 'search_rebuild',
+                                                                  'key': 'search_rebuild'})
+        if existing_task.get('state') == 'pending':
+            log.info('A pending task (%s) was found for this %s (%s). Skipping this duplicate task...',
+                     existing_task['id'], 'Organization' if is_org else 'Group', group.name)
+            return False
+        task['id'] = existing_task['id']
+    except NotFound:
+        pass
+
+    _get_action('task_status_update')({'session': model.meta.create_local_session(), 'ignore_auth': True}, task)
+
+    dataset_ids = [dataset.id for dataset in results]
+    job_title = 'Rebuild Dataset Indecies for %s: %s' % ('Organization' if is_org else 'Group', group.name)
```

Comment:
I guess job titles don't need to be translatable?

Reply:
I think the titles do show in job_list action. So might need translating? that being said, I don't think you want to translate it when making the Job, as it would go into REDIS with that..unless if gettext does reverse translations?

Reply:
I don't know that it can be solved here. If an existing text field mixes user data with English we don't have much hope of translating them. The job title could be just `'Rebuild Dataset Indices'` with the specific org left out of the message, then we could offer translations if these are displayed in an admin interface somewhere.

Reply:
yeah let's just do that then, keep it simple. The job and task info will have the argument with the group ID and the entity type, so don't need to specify in a human readable title too.